matrix/README.md

```diff
@@ -0,0 +1,109 @@
+![charmory logo](../docs/assets/charmory.png)
+
+---
+
+# Overview
+
+Armory-matrix provides a utility to automatically expand parameter
+specifications in order to perform a series of evaluations covering all possible
+combinations of parameters.
+
+# Installation
+
+```bash
+pip install armory-matrix
+```
+
+# Usage
+
+Use the `matrix` decorator to specify the ranges of values for all parameters of
```

Comment:
Ooh, this was a good question. Turns out because generator functions are callable, they get treated like dynamic parameters--so they do work, you just have to account for the fact they might get passed arguments. I thought I was testing this with `range`, but that's actually an object with a `__iter__`.

Reply:
Actually it's a bit more complicated than that. For example, using my own `frange` generator
```
@matrix(a=[1, 2], b=frange(1.0, 2.0, 0.5))
def foo(a, b):
  print((a, b))
```
only ends up running
a | b
--|--
1 | 1.0
2 | 1.5

because there's no way to reset a generator.